xonsh/readline_shell.py

```diff
@@ -121,6 +121,12 @@ def setup_readline():
             # this seems to fail with libedit
             print_exception('xonsh: could not load readline default init file.')
 
+    # handle case sensitive, see Github issue #1342 for details
+    if env.get('CASE_SENSITIVE_COMPLETIONS'):
+        readline.parse_and_bind("set completion-ignore-case off")
+    else:
+        readline.parse_and_bind("set completion-ignore-case on")
```

Comment:
This will work fine for startup, but will not work when dynamically setting the value at the command line.  Something more involved, like having a custom validator, or having this run before each prompt is needed to truly fix this bug.